ShellPkg/Library/UefiShellAcpiViewCommandLib/Parsers/Spmi/SpmiParser.c

```diff
@@ -0,0 +1,94 @@
+/** @file
+  SPMI table parser
+
+  Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+  @par Reference(s):
+    - IPMI 2.0 Specification - October 2013
+**/
+
+#include <IndustryStandard/Acpi.h>
+#include <Library/UefiLib.h>
+#include "AcpiParser.h"
+#include "AcpiTableParser.h"
+
+// Local variables
+STATIC ACPI_DESCRIPTION_HEADER_INFO  mAcpiHdrInfo;
+
+/**
+  Validate Interface Type.
+
+  @param [in] Ptr       Pointer to the start of the field data.
+  @param [in] Context   Pointer to context specific information e.g. this
+                        could be a pointer to the ACPI table header.
+**/
+STATIC
+VOID
+EFIAPI
+ValidateInterfaceType (
+  IN UINT8   *Ptr,
+  IN UINT32  Length,
+  IN VOID    *Context
+  )
+{
+  if (*Ptr > 5) {

```

Comment:
It would be better to have these values in the MdePkg:
```
0 Reserved
1 Keyboard Controller Style (KCS)
2 Server Management Interface Chip (SMIC)
3 Block Transfer (BT)
```

and use a macro instead of `5` here